.github/workflows/golangci-lint.yml

```diff
@@ -0,0 +1,23 @@
+name: golangci-lint
+on:
+  push:
+  pull_request:
+permissions:
+  contents: read
+  # Optional: allow read access to pull request. Use with `only-new-issues` option.
+  # pull-requests: read
+jobs:
+  golangci:
+    name: lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/setup-go@c4a742cab115ed795e34d4513e2cf7d472deb55f # v3.3.1
+        with:
+          go-version: 1.19
```

Comment:
The go-version here doesn't align with the one in go test.

Reply:
Bumped the go version to 1.20 across the repo. PTAL.

Reply:
Scratch that, I'll just make it 1.19 and 1..20 to be consistent. We'll bump the go version separately.